### DISCUSSION TITLE
Preserve electricity option in railroad mode

### DISCUSSION THREAD

**Discussion by Bakanych** [voxeltycoon/issues#539] [UI&UX]
Railroad electricity option is being reset to off every time when railroad menu is opened. It's really annoying when you're building electric railroad network. Can the option value be just preserved every time it's changed?

Thank you.

**Comment by victorleblais** [comment]
In addition to remembering if electricity was used or not, it should also remember which side of the tracks the poles are.

_In the following examples I'm assuming the default side for electric poles is to the left_

First example: _"Remember after closing"_

- Open Rail tools, then "Lay tracks"
- Activate electricity
- Change pole side to right
- Place some tracks
- Close the rail tools
- Re-open them
- → Electricity should be toggled on, with poles on the right

Second example: _"Remember after toggling"_

- Open Rail tools, then "Lay tracks"
- Activate electricity
- Change pole side to right
- Place some tracks
- Disable electricity
- Place some tracks
- Enable electricity
- → Electric poles should be on the right
